### DISCUSSION TITLE
OpENer as a CIP exchange initiator

### DISCUSSION THREAD

**Discussion by slayerserg** [EIPStackGroup/OpENer#516] [Q&A]
Hello,
Thanks for your work, it awesome such open source projects are exist! I would like to clarify one important thing for me: could OpENer be an initiator of CIP exchange? I need to connect OpENer with the other device that is waiting for connection on a TCP socket, is it possible for OpENer to specify the IP address to connect to and configure its messages to exchange or it's able only wait for incoming connection request?

**Comment by azoitl** [comment]
OpENer is per default a passive device. What you would need is in EtherNet/IP terms a scanner. OpENer currently has no scanner functionality. Depending on how much functionality you need and what exactly you would like to do you could use some of the OpENer infrastructure to build a kind of scanner. However for that you would need a rather deep understanding of the CIP and EtherNet/IP protocol.
